### PR TITLE
Redact query errors

### LINKED ISSUE #759: Return minimal info on errors returned to contracts
When discussing what information that can be returned in Interchain Accounts, we realized that [error messages and events are not guaranteed to be the same on every system](https://github.com/cosmos/ibc-go/issues/701#issuecomment-1018484522) (they are not part of consensus).

We can safely return data and *error code*. They have changed events in the sdk in minor upgrades, but for the same version, it will be the same on all systems (regardless of build system).

I would start limiting what we return to the contract and for the error returned in the reply section, not include error strings, rather the error code. This will make it harder to debug, but it will remove an attack vector that has popped up unexpectedly time after time. (Not in attacks, but in issues with different build systems)

@webmaster128 what do you think about this? I know this was also a concern of yours for windows and arm64 builds.

@alpe I wonder if there is a subset of info we could safely whitelist, like messages generated from x/wasm itself that don't include strings that come from other parts of the sdk.

### PR DESCRIPTION
**Pull request by ethanfrey**
Follow up from #765 

Closes #759

### CODE REVIEW

**Review by alpe — APPROVED** [review]
LGTM 👍 
The error path was covered by a test already

File: x/wasm/keeper/query_plugins.go [hunk 1]
```diff
@@ -60,6 +60,12 @@ func (q QueryHandler) Query(request wasmvmtypes.QueryRequest, gasLimit uint64) (
 	if ok := errors.As(err, &noSuchContract); ok {
 		return res, wasmvmtypes.NoSuchContract{Addr: noSuchContract.Addr}
 	}
+	// Otherwise redact all (we can theoretically redact less in the future)
+	if err != nil {
+		// Issue #759 - we don't return error string for worries of non-determinism
+		// moduleLogger(ctx).Info("Redacting submessage error", "cause", err)
```
Comment (larry0x) [inline x/wasm/keeper/query_plugins.go:66]: @alpe @webmaster128 Is it possible to have this error being logged?

Comment (alpe) [reply x/wasm/keeper/query_plugins.go:66]: No problem. I have opened an issue to track the work

Comment (webmaster128) [reply x/wasm/keeper/query_plugins.go:66]: https://github.com/CosmWasm/wasmd/issues/1593

Comment (larry0x) [reply x/wasm/keeper/query_plugins.go:66]: Thank you

**Pull request merged**